download-utils/src/lib.rs

```diff
@@ -108,17 +110,79 @@ pub fn download_file(url: &str, destination_file: &Path) -> Result<(), String> {
     Ok(())
 }
 
-fn extract_archive(archive_filename: &Path, destination_dir: &Path) -> Result<(), String> {
+const MAXIMUM_GENESIS_ARCHIVE_SIZE: u64 = 10 * 1024 * 1024; // 10 MiB
+
+fn is_valid_archive_entry(parts: &[&str], kind: tar::EntryType) -> bool {
```

Comment:
Perhaps this:
```suggestion
fn is_valid_genesis_archive_entry(parts: &[&str], kind: tar::EntryType) -> bool {
```
as a snapshot archive has a different layout